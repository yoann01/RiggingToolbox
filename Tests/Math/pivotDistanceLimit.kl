
require RiggingToolbox;

operator entry(){

  Xfo inpivotXfo(Vec3(0.0, 10.0, 0.0));
  Scalar pivotLength = 5.0;
  Vec3 intarget(0.0, 0.0, 0.0);
  Vec3 pivotLocalAxis(1.0, 0.0, 0.0);

  Scalar inVal = 3.0;
  Vec2 minValSoftening(2.0, 3.0);
  Scalar maxVal = 15.0;
  Vec2 maxValSoftening(3.0, 5.0);

  Scalar minDistLength = 6.0;
  Vec2 minDistLengthSoftening(2.0, 3.0);
  Scalar maxDistLength = 12.0;
  Vec2 maxDistLengthSoftening(3.0, 5.0);
  Scalar maxDeltaAngle = 0.5;
  Scalar maxDeltaAngleSoftening = 0.1;

  for(Integer i=2; i<20; i++){
    inpivotXfo.tr.y = Scalar(i);
    Xfo iopivotXfo = inpivotXfo;
    Vec3 iotarget = intarget;
    Scalar outPivotDist = pivotDistanceLimit(
      iopivotXfo,
      pivotLength,
      iotarget,
      pivotLocalAxis,
      minDistLength,
      minDistLengthSoftening,
      maxDistLength,
      maxDistLengthSoftening,
      maxDeltaAngle, 
      maxDeltaAngleSoftening
      );
    report("inpivotXfo:" + inpivotXfo + " iopivotXfo:" + iopivotXfo + " outPivotDist:" + outPivotDist + " iotarget:" + iotarget);
  }
}

