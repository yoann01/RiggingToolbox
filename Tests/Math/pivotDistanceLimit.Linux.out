inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+2.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.515264},w:+0.857031},tr:{x:+0.0,y:+2.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+6.831099 iotarget:{x:+2.384185e-7,y:+0.476837e-6,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+3.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.284978},w:+0.958533},tr:{x:+0.0,y:+3.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+7.098571 iotarget:{x:+0.476837e-6,y:+0.476837e-6,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+4.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.182851},w:+0.98314},tr:{x:+0.0,y:+4.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+7.441874 iotarget:{x:+0.0,y:+0.0,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+5.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.11585},w:+0.993266},tr:{x:+0.0,y:+5.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+7.842641 iotarget:{x:+0.0,y:+0.0,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+6.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.063966},w:+0.997952},tr:{x:+0.0,y:+6.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+8.28615 iotarget:{x:-0.476837e-6,y:-0.476837e-6,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+7.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+1.973274e-2},w:+0.999805},tr:{x:+0.0,y:+7.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+8.761395 iotarget:{x:+0.0,y:+0.0,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+8.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.031541},w:+0.999502},tr:{x:+0.0,y:+8.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+9.162743 iotarget:{x:+0.476837e-6,y:+0.0,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+9.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.065739},w:+0.997836},tr:{x:+0.0,y:+9.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+9.485862 iotarget:{x:+0.112063,y:+0.188642,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+10.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.087925},w:+0.996127},tr:{x:+0.0,y:+10.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+9.817627 iotarget:{x:+0.261044,y:+0.483841,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+11.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.109811},w:+0.993952},tr:{x:+0.0,y:+11.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+10.15614 iotarget:{x:+0.392595,y:+0.797235,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+12.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.131936},w:+0.991258},tr:{x:+0.0,y:+12.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+10.5 iotarget:{x:+0.506343,y:+1.121838,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+13.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.154858},w:+0.987936},tr:{x:+0.0,y:+13.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+10.84814 iotarget:{x:+0.601984,y:+1.450533,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+14.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.17929},w:+0.983796},tr:{x:+0.0,y:+14.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+11.19977 iotarget:{x:+0.678674,y:+1.774985,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+15.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.206375},w:+0.978472},tr:{x:+0.0,y:+15.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+11.55427 iotarget:{x:+0.734065,y:+2.083182,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+16.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.238527},w:+0.971135},tr:{x:+0.0,y:+16.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+11.91114 iotarget:{x:+0.761549,y:+2.351751,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+17.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.247403},w:+0.968912},tr:{x:+0.0,y:+17.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+12.0 iotarget:{x:+0.934647,y:+3.110485,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+18.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.247403},w:+0.968912},tr:{x:+0.0,y:+18.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+12.0 iotarget:{x:+1.139236,y:+4.050985,z:+0.0}
inpivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+0.0,y:+19.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} iopivotXfo:{ori:{v:{x:+0.0,y:+0.0,z:-0.247403},w:+0.968912},tr:{x:+0.0,y:+19.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}} outPivotDist:+12.0 iotarget:{x:+1.32175,y:+5.001207,z:+0.0}
